examples/Bickley_jet_shallow_water.jl

```diff
@@ -0,0 +1,244 @@
+# the evolution of an unstable, geostrophically balanced, Bickley jet.
+# The model solves the governing equations for the shallow water model in
+# conservation form.  The geometry is that of a periodic channel
+# in the ``x``-direction with a flat bottom and a free-surface.  The initial
+# conditions are that of a Bickley jet with small amplitude perturbations.
+# The interested reader can see ["The nonlinear evolution of barotropically unstable jets," J. Phys. Oceanogr. (2003)](https://doi.org/10.1175/1520-0485(2003)033<2173:TNEOBU>2.0.CO;2)
+# for more details on this specific problem. 
+#
+# Unlike the other models, the fields that are simulated are the mass transports, 
+# ``uh`` and ``vh`` in the ``x`` and ``y`` directions, respectively,
+# and the height ``h``.  Note that ``u`` and ``v`` are the velocities, which 
+# can easily be computed when needed.
+#
+# ## Install dependencies
+#
+# First we make sure that we have all of the packages that are required to
+# run the simulation.
+# ```julia
+# using Pkg
+# pkg"add Oceananigans, NCDatasets, Plots, Printf, Polynomials"
+# ```
+#
+
+using Oceananigans
+using Oceananigans.Models: ShallowWaterModel
+
+# ## Two-dimensional domain 
+#
+# The shallow water model is a two-dimensional model and we must specify
+# the number of vertical grid points to be one.  
+# We pick the length of the domain to fit one unstable mode along the channel
+# exactly.  Note that ``Lz`` is the mean depth of the fluid.  It is determined
+# using linear stability theory that the zonal wavenumber that yields the most unstable
+# mode that fits in this zonal channel has a growth rate of ``\sigma \approx 0.14``.
+
+Lx = 2π
+Ly = 20
+Lz = 1
+Nx = 128
+Ny = Nx
+
+grid = RegularRectilinearGrid(size = (Nx, Ny, 1),
+                            x = (0, Lx), y = (0, Ly), z = (0, Lz),
+                            topology = (Periodic, Bounded, Bounded))
+
+# ## Physical parameters
+#
+# This is a toy problem and we choose the parameters so this jet idealizes
+# a relatively narrow mesoscale jet.   
+# The physical parameters are
+#
+#   * ``f``: Coriolis parameter
+#   * ``g``: Acceleration due to gravity
+#   * ``U``: Maximum jet speed
+#   * ``\Delta\eta``: Maximum free-surface deformation that is dictated by geostrophy
+#   * ``\epsilon`` : Amplitude of the perturbation
+
+f = 1
+g = 9.80665
+U = 1.0
+Δη = f * U / g
+ϵ = 1e-4
+nothing # hide
+
+# ## Building a `ShallowWaterModel`
+#
+# We use `grid`, `coriolis` and `gravitational_acceleration` to build the model.
+# Furthermore, we specify `RungeKutta3` for time-stepping and `WENO5` for advection.
+
+model = ShallowWaterModel(
+    timestepper=:RungeKutta3,
+    advection=WENO5(),
+    grid=grid,
+    gravitational_acceleration=g,
+    coriolis=FPlane(f=f),
+    )
+
+# ## Background state and perturbation
+# 
+# We specify `Ω` to be the background vorticity of the jet in the absence of any perturbations.
```

Comment:
Before "We specify `Ω`..." perhaps we can write down the basic state in latex so that users have a sense why the `uⁱ`, `hⁱ`, ... fields are prescribed as such below?

Reply:
Also, doesn't the growth rate σ=0.14 depends on the base state? I suggest after stating what the base state is, we bring down the remark about the theoretical value of the growth rate. E.g.,

```julia
# Using linear stability theory we can predict that the zonal wavenumber corresponding
# to the most unstable mode for this basic state and fits in our domain grows at a rate 
# of ``\sigma \approx 0.14``.
```

Reply:
Even better simply say `of approximately 0.14.` and avoid introducing a new symbol `σ` that might never be used again (if you end up agreeing with all my suggestions below).

Reply:
All good suggestions.  I have made changes but will let you read them over before we consider this resolved.

Reply:
Agreed